File: app/src/main/java/com/example/totoroto/mureok/Main/MainActivity.java
package com.example.totoroto.mureok.Main;

import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.design.widget.NavigationView;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.GravityCompat;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.example.totoroto.mureok.Info.InfoActivity;
import com.example.totoroto.mureok.Login.LoginActivity;
import com.example.totoroto.mureok.Manage.ManageFragment;
import com.example.totoroto.mureok.R;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import de.hdodenhof.circleimageview.CircleImageView;

public class MainActivity extends AppCompatActivity {
    private final String TAG = "SOLBIN";

    private TabLayout tabLayout;
    private ViewPager viewPager;
    private Toolbar toolbar;
    private NavigationView navigationView;
    private DrawerLayout drawerLayout;
    private TextView tvNickName;
    private TextView tvEmail;
    private CircleImageView civProfilePicture;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        init();
        aboutToolbar();
        aboutTab();
        aboutNavi();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        for (Fragment fragment : getSupportFragmentManager().getFragments()) {
            try {
                fragment.onActivityResult(requestCode, resultCode, data);
            }catch (Exception e){
                e.printStackTrace();
            }
        }
    }

    private void aboutNavi() {

        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                item.setChecked(true);
                drawerLayout.closeDrawers();

                switch (item.getItemId()) {
                    case R.id.navi_item_shares:
                       // moveInfoShareActivity();
                        break;
                    case R.id.navi_item_myInfo:
                        moveInfoActivity();
                        break;
                    case R.id.navi_item_send_opinion:
                        break;
                    case R.id.navi_item_logout:
                        aboutLogout();
                        break;
                }
                return true;
            }
        });
    }


    private void aboutLogout() {
        FirebaseAuth fAuth = FirebaseAuth.getInstance();
        fAuth.signOut();
        Intent intent = new Intent(getApplicationContext(), LoginActivity.class);
        startActivity(intent);
        Toast.makeText(getApplicationContext(), "로그아웃 되었습니다.", Toast.LENGTH_SHORT).show();
    }

    private void moveInfoActivity() {
        Intent intent = new Intent(getApplicationContext(), InfoActivity.class);
        startActivity(intent);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            drawerLayout.openDrawer(GravityCompat.START);

            //navigation header
            FirebaseUser fUser = FirebaseAuth.getInstance().getCurrentUser();
            if(fUser != null) {
                tvEmail.setText(fUser.getEmail());
                tvNickName.setText(fUser.getDisplayName());
                Glide.with(getApplicationContext())
                        .load(fUser.getPhotoUrl())
                        .centerCrop()
                        .diskCacheStrategy(DiskCacheStrategy.SOURCE)
                        .into(civProfilePicture);
            }
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private void aboutToolbar() {
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        actionBar.setHomeAsUpIndicator(R.drawable.ic_drawer);
        actionBar.setDisplayHomeAsUpEnabled(true);
    }

    private void aboutTab() {
        //관리 화면을 첫화면으로
        getSupportFragmentManager().beginTransaction().replace(R.id.mainFrameLayout, ManageFragment.newInstance()).commit();

        //탭 추가
        tabLayout.addTab(tabLayout.newTab().setIcon(R.drawable.selector_tab_manage));
        tabLayout.addTab(tabLayout.newTab().setIcon(R.drawable.selector_tab_gallery));
        tabLayout.addTab(tabLayout.newTab().setIcon(R.drawable.selector_tab_community));
        tabLayout.setTabGravity(TabLayout.GRAVITY_FILL); //너비를 모두 같게 표시


        TabAdapter tabAdapter = new TabAdapter(getSupportFragmentManager());
        viewPager.setAdapter(tabAdapter);
        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));

        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewPager.setCurrentItem(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {
            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {
            }
        });
    }

    private void init() {
        tabLayout = (TabLayout) findViewById(R.id.tabLayout);
        viewPager = (ViewPager)findViewById(R.id.viewPager);
        toolbar = (Toolbar) findViewById(R.id.toolbar);
        navigationView = (NavigationView) findViewById(R.id.navigationView);
        drawerLayout = (DrawerLayout) findViewById(R.id.drawerLayout);

        View header = navigationView.getHeaderView(0);
        tvNickName = (TextView) header.findViewById(R.id.tvProfile_navi);
        tvEmail = (TextView) header.findViewById(R.id.tvEmail_navi);
        civProfilePicture = (CircleImageView) header.findViewById(R.id.civProfile_navi);
    }
}

File: app/src/main/java/com/example/totoroto/mureok/main/MainActivity.kt
package com.example.totoroto.mureok.main

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.view.GravityCompat
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import android.widget.Toast
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.example.totoroto.mureok.R
import com.example.totoroto.mureok.info.InfoActivity
import com.example.totoroto.mureok.login.LoginActivity
import com.example.totoroto.mureok.manage.ManageFragment
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.drawer_header.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        aboutToolbar()
        aboutTab()
        aboutNavi()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        super.onActivityResult(requestCode, resultCode, data)
        for (fragment in supportFragmentManager.fragments) {
            try {
                fragment.onActivityResult(requestCode, resultCode, data)
            } catch (e: Exception) {
                e.printStackTrace()
            }

        }
    }

    private fun aboutNavi() {

        navigationView.setNavigationItemSelectedListener { item ->
            item.isChecked = true
            drawerLayout.closeDrawers()

            when (item.itemId) {
                R.id.navi_item_myInfo -> moveInfoActivity()
                R.id.navi_item_send_opinion -> {
                }
                R.id.navi_item_logout -> aboutLogout()
            }
            true
        }
    }


    private fun aboutLogout() {
        val fAuth = FirebaseAuth.getInstance()
        fAuth.signOut()
        val intent = Intent(applicationContext, LoginActivity::class.java)
        startActivity(intent)
        Toast.makeText(applicationContext, "로그아웃 되었습니다.", Toast.LENGTH_SHORT).show()
    }

    private fun moveInfoActivity() {
        val intent = Intent(applicationContext, InfoActivity::class.java)
        startActivity(intent)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            drawerLayout.openDrawer(GravityCompat.START)

            //navigation header
            val fUser = FirebaseAuth.getInstance().currentUser
            if (fUser != null) {
                tvEmail_navi.text = fUser.email
                tvProfile_navi.text = fUser.displayName
                Glide.with(applicationContext)
                        .load(fUser.photoUrl)
                        .centerCrop()
                        .diskCacheStrategy(DiskCacheStrategy.SOURCE)
                        .into(civProfile_navi)
            }
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    private fun aboutToolbar() {
        setSupportActionBar(toolbar)
        val actionBar = supportActionBar ?: return
        actionBar.setHomeAsUpIndicator(R.drawable.ic_drawer)
        actionBar.setDisplayHomeAsUpEnabled(true)
    }

    private fun aboutTab() {
        //관리 화면을 첫화면으로
        supportFragmentManager.beginTransaction().replace(R.id.mainFrameLayout, ManageFragment.newInstance()).commit()

        //탭 추가
        tabLayout.addTab(tabLayout.newTab().setIcon(R.drawable.selector_tab_manage))
        tabLayout.addTab(tabLayout.newTab().setIcon(R.drawable.selector_tab_gallery))
        tabLayout.addTab(tabLayout.newTab().setIcon(R.drawable.selector_tab_community))
        tabLayout.tabGravity = TabLayout.GRAVITY_FILL //너비를 모두 같게 표시


        val tabAdapter = TabAdapter(supportFragmentManager)
        viewPager.adapter = tabAdapter
        viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabLayout))

        tabLayout.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                viewPager.currentItem = tab.position
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {}

            override fun onTabReselected(tab: TabLayout.Tab) {}
        })
    }

    companion object {
        const val TAG = "SOLBIN"
    }
}
